src/constraint_system/ecc/scalar_mul/fixed_base.rs

```diff
@@ -26,139 +27,144 @@ fn compute_wnaf_point_multiples(
     dusk_jubjub::batch_normalize(&mut multiples).collect()
 }
 
-/// Computes a BlsScalar multiplication with the input scalar and a chosen
-/// generator
-pub fn scalar_mul(
-    composer: &mut StandardComposer,
-    jubjub_scalar: Variable,
-    generator: JubJubExtended,
-) -> PointScalar {
-    // XXX: we can slice off 3 bits from the top of wnaf, since F_r prime has
-    // 252 bits. XXX :We can also move to base4 and have half the number of
-    // gates since wnaf adjacent entries product is zero, we will not go over
-    // the specified amount
-    let num_bits = 256;
-
-    // compute 2^iG
-    let mut point_multiples = compute_wnaf_point_multiples(generator, num_bits);
-    point_multiples.reverse();
-
-    // Fetch the raw scalar value as bls scalar, then convert to a jubjub scalar
-    // XXX: Not very Tidy, impl From function in JubJub
-    let raw_bls_scalar = composer.variables.get(&jubjub_scalar).unwrap();
-    let raw_jubjub_scalar =
-        JubJubScalar::from_bytes(&raw_bls_scalar.to_bytes()).unwrap();
-
-    // Convert scalar to wnaf_2(k)
-    let wnaf_entries = raw_jubjub_scalar.compute_windowed_naf(2);
-    assert_eq!(wnaf_entries.len(), num_bits);
-
-    // Initialise the accumulators
-    let mut scalar_acc: Vec<BlsScalar> = Vec::new();
-    scalar_acc.push(BlsScalar::zero());
-    let mut point_acc: Vec<JubJubAffine> = Vec::new();
-    point_acc.push(JubJubAffine::identity());
-
-    // Auxillary point to help with checks on the backend
-    let mut xy_alphas = Vec::new();
-
-    // Load values into accumulators based on wnaf entries
-    for (i, entry) in wnaf_entries.iter().rev().enumerate() {
-        // Based on the WNAF, we decide what scalar and point to add
-        let (scalar_to_add, point_to_add) = match entry {
+impl StandardComposer {
```

Comment:
Please, also make available the already existing `zero_var` witness as a composer method.

This is a very special number and is used in 99% of the circuits. That's why it should be treated differently.

Reply:
This should be a sepparated issue, and something that does not concern to this PR. Also, we discussed that with kev and considered that would make the circuits confusing. Since the same zero would be used as input, output, left or right wire value etc.. It doesn;t make sense.

Reply:
Again; zero is a special number and should be treated as exception. As described on the issue below, every circuit / gadget loses an additional gate for something that can be trivially exposed by the API

https://github.com/dusk-network/plonk/issues/526